design/events.md

```diff
@@ -370,19 +367,19 @@ The system does not provide any ordering guarantees. Events may be emitted in an
 
 ### 2.3.3. Delivery contract
 
-Event delivery is the act of handing over an event to the subscriber's `EventListener`. For optimal event delivery,
-implementers of the `EventListener` interface are expected to respect the following requirements:
+Event delivery is the act of handing over an event to the `EventSubscriber`. For optimal event delivery, implementers of
+the `EventSubscriber` interface are expected to respect the following requirements:
 
-* Methods in the `EventListener` interface MUST NOT block the calling thread. If required, implementers of the
-  `EventListener` interface should rather dispatch the calls to separate thread pools, if the underlying event
+* `onXyz()` methods in the `EventSubscriber` interface MUST NOT block the calling thread. If required, implementers of
+  the `EventSubscriber` interface should rather dispatch the calls to separate thread pools, if the underlying event
   processing requires blocking I/O.
-* Methods in the `EventListener` interface operate in fire and forget mode: the method MUST return as soon as the
-  should not wait for read acknowledgements.
-* The server COULD impose a (configurable) timeout after which the thread calling the `EventListener` method is
-  interrupted and delivery is considered failed.
+* `onXyz()` methods in the `EventSubscriber` interface operate in fire and forget mode: the method MUST return
+  immediately; it should not wait for read acknowledgements from remote systems.
+* The server COULD impose a (configurable) timeout after which the thread calling the `EventSubscriber` `onXyz()` method
+  is interrupted and delivery is considered failed.
 * A `RuntimeException` MAY be thrown to indicate that event delivery wasn't possible, e.g. because the underlying
   pub/sub system is unreachable. But in this case, the exception MUST be thrown immediately.
```

Comment:
off topic to this PR, but I think this contradicts the requirements that `onXyz()` methods must not do blocking I/O on the calling thread. How can they immediately determine whether the even delivery was possible without blocking? If the intention is to allow subscribers to report asynchronously determined problems as exceptions on the notification thread, each call still cannot reliably throw if there's an error. What is the benefit of specifying this failure-handling behaviour? Could we simply say that subscribers MAY throw `RuntimeExceptions` without going into details?

From my POV we might as well add that subscribers SHOULD NOT throw `RuntimeExceptions` if they can avoid it. What can Nessie do about those exceptions, except perhaps log them?

Reply:
I agree, this doesn't make a lot of sense :-) I will simplify as you suggest.